ios/brave-ios/Sources/Brave/Frontend/Sync/SyncAddDevice/SyncAddDeviceViewController.swift

```diff
@@ -0,0 +1,281 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import BraveCore
+import BraveShared
+import BraveUI
+import BraveWallet
+import Data
+import Shared
+import UIKit
+
+enum SyncDeviceType {
+  case mobile
+  case computer
+}
+
+class SyncAddDeviceViewController: SyncViewController {
+
+  // MARK: UX
+
+  private var scrollViewContainer = UIScrollView()
+
+  private var contentStackView = UIStackView().then {
+    $0.axis = .vertical
+    $0.spacing = 20
+  }
+
+  private var modeControl = UISegmentedControl(items: [Strings.QRCode, Strings.codeWords]).then {
+    $0.translatesAutoresizingMaskIntoConstraints = false
+    $0.selectedSegmentIndex = 0
+    $0.setContentCompressionResistancePriority(.defaultHigh, for: .vertical)
+    $0.selectedSegmentTintColor = UIColor(braveSystemName: .containerBackground)
+    $0.setTitleTextAttributes(
+      [.foregroundColor: UIColor(braveSystemName: .textPrimary)],
+      for: .selected
+    )
+  }
+
+  private let titleDescriptionStackView = SyncAddDeviceInformationView()
+
+  private let codeDetailsStackView: SyncAddDeviceCodeView
+
+  private let codeExpirationStackView = SyncAddDeviceCodeExpirationView()
+
+  private let actionButtonStackView = SyncAddDeviceActionView()
+
+  // MARK: Internal
+
+  private var pageTitle: String = Strings.sync
+
+  private var deviceType: SyncDeviceType = .mobile
+
+  private var isSyncCodeExpired = false {
+    didSet {
+      changeCodeDisplayStatus()
+    }
+  }
+
+  private var isSyncCodePresented: Bool {
+    modeControl.selectedSegmentIndex == 0
+  }
+
+  private let syncAPI: BraveSyncAPI
+
+  var addDeviceHandler: (() -> Void)?
+
+  // MARK: Lifecycle
+
+  init(title: String, type: SyncDeviceType, syncAPI: BraveSyncAPI) {
+    self.syncAPI = syncAPI
+    codeDetailsStackView = SyncAddDeviceCodeView(syncAPI: syncAPI)
+
+    super.init()
+
+    resetExpirationDate()
+
+    pageTitle = title
+    deviceType = type
+  }
+
+  @available(*, unavailable)
+  required init(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    setTheme()
+    doLayout()
+
+    enableDeviceType()
+  }
+
+  override func viewDidAppear(_ animated: Bool) {
+    super.viewDidAppear(animated)
+
+    if !syncAPI.isInSyncGroup && !syncAPI.isSyncFeatureActive
+      && !syncAPI.isInitialSyncFeatureSetupComplete
+    {
+      showInitializationError()
+    }
+  }
+
+  override func viewDidLayoutSubviews() {
+    super.viewDidLayoutSubviews()
+
+    scrollViewContainer.contentSize = CGSize(
```

Comment:
`scrollViewContainer.contentSize = contentStackView.frame.size`?

Reply:
Nice catch no need to do it.

Fix at https://github.com/brave/brave-core/pull/23394/commits/51acf862c997441047ace17712c0fc80e726ca75